### ISSUE TITLE
Movielens benchmark

### ISSUE THREAD

**Issue by bkj** [RankSys/RankSys#54]
At 27:15 in [this video](https://www.youtube.com/watch?v=AQ0EjOJnu9Q), the speaker mentions that @saulvargas provided some benchmarks for a "binary prediction of a random 80/20 split of the MovieLens 20M dataset".  

Does anyone here know exactly what he means?  Eg, how the split was done, how the popularity baseline works, how the matrix factorization baseline works, etc?

Thanks!
Ben

**Comment by saulvargas** [comment]
More details here: https://github.com/amzn/amazon-dsstne/issues/24
I tried to benchmark DSSTNE with some of the algorithms in RankSys, but I obtained very poor performance with the former. The author of DSSTNE claims he did some fixes and managed to improve on my results, but I have not been able to replicate that.

**Comment by bkj** [comment]
Great -- thanks for the pointer.  Will post here if I manage to figure anything out.

**Comment by bkj** [comment]
Also -- I know this was a long time ago, but... do you know how you got the
```
pop 0.10974162112149495
```
number? 

I'm getting ~0.085 -- but maybe I'm doing the split differently...

**Comment by saulvargas** [comment]
Are you running the script in https://github.com/saulvargas/dsstne-comparison/blob/master/run.sh ?

Otherwise, yes, slight variations in the split can justify this discrepancy. I would not worry too much about absolute values, but on relative ones.

**Comment by bkj** [comment]
I'm doing something similar, though not using that script.  I've been able to get >0.3 P@10 using the models posted in that thread -- since you weren't able to reproduce Scott's results, I'll have to dig into your code to see what we're doing different.

**Comment by bkj** [comment]
Here's some scripts that get 0.35 P@10: https://github.com/bkj/dsstne-scripts

Let me know if you have a chance to take a look, and if there's anything obviously wrong.

**Comment by saulvargas** [comment]
What about the baselines from RankSys in your setup?

To be honest, I have not used DSSTNE or RankSys in a while, nowadays I use Keras.

**Comment by bkj** [comment]
I haven't had time to look at the RankSys baselines specifically, but I've been able to get 0.25-0.28 using matrix factorization methods, which is in line w/ your results.

Makes sense to do these models in Keras -- though I bet it's substantially slower than DSSTNE.  I used `pytorch` to implement a simple fully connected network to imitate the [DSSTNE model](https://github.com/bkj/dsstne-scripts/blob/master/config.json) and it's ~100x slower.  Would be interesting to hear how Keras stacks up performance-wise.
